pkg/models/dashboards_public.go

```diff
@@ -0,0 +1,43 @@
+package models
+
+var (
+	ErrPublicDashboardFailedGenerateUniqueUid = DashboardErr{
+		Reason:     "Failed to generate unique dashboard id",
+		StatusCode: 500,
+	}
+	ErrPublicDashboardNotFound = DashboardErr{
+		Reason:     "Public dashboard not found",
+		StatusCode: 404,
+		Status:     "not-found",
+	}
+	ErrPublicDashboardIdentifierNotSet = DashboardErr{
+		Reason:     "No Uid for public dashboard specified",
+		StatusCode: 400,
+	}
+)
+
+type PublicDashboardConfig struct {
+	IsPublic        bool            `json:"isPublic"`
+	PublicDashboard PublicDashboard `json:"publicDashboard"`
+}
```

Comment:
In the future this will be:
```

type PublicDashboardConfig struct {
	IsPublic         bool              `json:"isPublic"`
	PublicDashboards []PublicDashboard `json:"publicDashboard"`
}
```

Reply:
Ah, I understand. it's a bit confusing given that model that IsPublic is really part of the dashboard entity. 

I thought it would be something like this (if we want to support multiple public dashboards with different settings). Originally I did not think about supporting multiple with different settings. Just same settings but different access keys for different users (When shared via emails). But I can see the value in supporting different settings as well, especially for variables.

````ts
table dashboard_public  {
  isPublic: bool
  uid: string;
  dashboardUid: string
  settings:  {
      timeRange: { canChange: bool, ... }, 
      variables: {.... }         
  }
} 

table dashboard_public_access_key  {
  publicDashbordUid: string;
  key: string;  
  userEmail?: string;
  settings?:  {
      timeRange: { canChange: bool, ... }, 
      variables: {.... }         
  }
}
```

But your model also works ok I think, would have to duplicate the settings for each user email but the backend could help with that, but then again the frontend would be messy showing each email you share it with as a separate public dashboard